wormhole-connect/src/views/v2/Redeem/index.tsx

```diff
@@ -610,14 +636,23 @@ const Redeem = () => {
         config.tokens[receivedTokenKey],
         'Solana',
       );
-      const ata = getAssociatedTokenAddressSync(
-        new PublicKey(receivedToken.address.toString()),
-        new PublicKey(receivingWallet.address),
-      );
-      if (!ata.equals(new PublicKey(recipient))) {
-        setClaimError('Not connected to the receiving wallet');
+
+      try {
+        const ata = getAssociatedTokenAddressSync(
+          new PublicKey(receivedToken.address.toString()),
+          new PublicKey(receivingWallet.address),
+        );
+        if (!ata.equals(new PublicKey(recipient))) {
+          setClaimError('Not connected to the receiving wallet');
+          return false;
+        }
+      } catch (e: unknown) {
+        console.log(
+          `Error while checking associated token address for the recipient: ${e}`,
```

Comment:
When an EVM wallet is connected as dest, the above call to getAssociatedTokenAddressSync throws "Non-base58 character" error. In that case we log and return false. I was thinking maybe we can also do `setClaimError('Not connected to the receiving wallet')`, but I wasn't sure if that's the only case we get this error for.

Reply:
isn't it a bug that we can be in that state to begin with? if `toChain` is `Solana`, we shouldn't be able to have an EVM wallet connected in the first place.

Reply:
This happens when user navigates from the widget to Redeem view; where toChain can be anything previously selected by the user.

Reply:
ok. so we probably need code that looks for this state change, and auto-connects the correct wallet.

Reply:
opened a separate issue to track this:
https://github.com/wormhole-foundation/wormhole-connect/issues/2930